idea/src/org/jetbrains/kotlin/idea/intentions/ConvertTwoComparisonsToRangeCheckIntention.kt

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2010-2016 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.kotlin.idea.intentions
+
+import com.intellij.openapi.editor.Editor
+import org.jetbrains.kotlin.builtins.KotlinBuiltIns
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.idea.intentions.branchedTransformations.evaluatesTo
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.resolve.calls.callUtil.getType
+
+class ConvertTwoComparisonsToRangeCheckIntention : SelfTargetingOffsetIndependentIntention<KtBinaryExpression>(KtBinaryExpression::class.java, "Convert to range check") {
+
+    private data class RangeExpressionData(val value: String, val min: String, val max: String)
+
+    override fun isApplicableTo(condition: KtBinaryExpression) = generateRangeExpressionData(condition) != null
+
+    override fun applyTo(condition: KtBinaryExpression, editor: Editor?) {
+        val rangeData = generateRangeExpressionData(condition) ?: return
+        condition.replace(KtPsiFactory(condition).createExpressionByPattern("$0 in $1..$2", rangeData.value, rangeData.min, rangeData.max))
+    }
+
+    private fun generateRangeExpressionData(condition: KtBinaryExpression): RangeExpressionData? {
+        if (condition.operationToken != KtTokens.ANDAND) return null
+        val firstCondition = condition.left as? KtBinaryExpression ?: return null
+        val secondCondition = condition.right as? KtBinaryExpression ?: return null
+        val firstOpToken = firstCondition.operationToken
+        val secondOpToken = secondCondition.operationToken
+        val firstLeft = firstCondition.left ?: return null
+        val firstRight = firstCondition.right ?: return null
+        val secondLeft = secondCondition.left ?: return null
+        val secondRight = secondCondition.right ?: return null
+
+        return when (firstOpToken) {
```

Comment:
This `when` is written in a very straight-forward way. Please, think a bit how to make it shorter. All branches are very like one another.